xcdat/axis.py

```diff
@@ -470,6 +475,58 @@ def _get_prime_meridian_index(lon_bounds: xr.DataArray) -> np.ndarray | None:
     return p_meridian_index
 
 
+def _adjust_bounds_for_prime_meridian(
+    ds: xr.Dataset, key: str, p_meridian_index: np.ndarray
+) -> xr.Dataset:
+    """Adjusts bounds for the prime meridian cell in longitude coordinates.
+
+    This function modifies the longitude bounds to handle cases where a grid
+    cell spans the prime meridian. It ensures that the longitude bounds are
+    aligned to the (0, 360) range by splitting the prime meridian cell into
+    two parts and adjusting the bounds accordingly. The extra cell created
+    during this process is removed to restore the original shape of the dataset.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        The Dataset containing longitude bounds and coordinates.
+    key : str
+        The name of the longitude bounds variable in the dataset.
+    p_meridian_index : np.ndarray
+        An array with a single element representing the index of the prime
+        meridian cell.
+
+    Returns
+    -------
+    xr.Dataset
+        The Dataset with adjusted longitude bounds for the prime meridian cell.
+    """
+    ds_new = ds.copy()
+
+    ds_new = _align_lon_to_360(ds_new, ds_new[key], p_meridian_index)
+    dim = get_dim_keys(ds_new[key], "X")
+
+    # NOTE: The logic for calculating new bounds is similar to the
+    # implementation in xcdat.bounds.create_bounds.
+    # Calculate the bounds interval for the prime meridian cell using
+    # the cell as the midpoint and the difference of the bounds.
+    bounds = ds_new[key].isel({dim: 0})
+    bounds_dim = _get_bounds_dim(ds_new[dim], ds_new[key])
+    bounds_delta = abs(bounds.diff(bounds_dim).item())
+
+    # Create new bounds for the prime meridian cell.
+    midpoint = ds_new[dim][p_meridian_index].item()
+    new_lower = midpoint - bounds_delta
+    new_upper = new_lower + bounds_delta
+
+    ds_new[key][p_meridian_index, :] = [new_lower, new_upper]
+
+    # Remove the extra cell to restore the original shape.
+    ds_new = ds_new.isel({dim: slice(0, -1)})
+
+    return ds_new
```

Comment:
The main function to analyze.

Reply:
I just made a fix for `    bounds_delta = abs(bounds.diff(bounds_dim).item()) / 2`.